tests/Unit/ControlSystem/Test_UpdateFunctionOfTime.cpp

```diff
@@ -0,0 +1,228 @@
+      t0, init_quat_func, init_pp_func, expr_time);
+
+  const DataVector updated_deriv{3, 1.0};
+
+  // Construct mock system and set to Testing phase
+  ActionTesting::MockRuntimeSystem<TestingMetavariables> runsys{
+      {}, {std::move(f_of_t_map)}};
+  ActionTesting::emplace_singleton_component<
+      TestSingleton<TestingMetavariables>>(make_not_null(&runsys),
+                                           ActionTesting::NodeId{0},
+                                           ActionTesting::LocalCoreId{0});
+  ActionTesting::set_phase(make_not_null(&runsys),
+                           TestingMetavariables::Phase::Testing);
+
+  // Update functions of time in global cache with new deriv
+  auto& cache =
+      ActionTesting::cache<TestSingleton<TestingMetavariables>>(runsys, 0_st);
+
+  for (auto& name : {pp_name, quatfot_name}) {
+    Parallel::mutate<
+        domain::Tags::FunctionsOfTime,
+        control_system::Mutations::update_function_of_time<deriv_order>>(
+        cache, name, update_time, updated_deriv, new_expiration_time);
+  }
+
+  // Update expected function of time
+  expected_pp.update(update_time, updated_deriv, new_expiration_time);
+  expected_quatfot.update(update_time, updated_deriv, new_expiration_time);
+
+  // Check that the FunctionsOfTime are what we expected
+  const auto& cache_f_of_t_map =
+      get<domain::Tags::FunctionsOfTime, TestingMetavariables>(cache);
+  const auto& cache_pp = dynamic_cast<
+      const domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>&>(
+      *(cache_f_of_t_map.at(pp_name)));
+  const auto& cache_quatfot = dynamic_cast<
+      const domain::FunctionsOfTime::QuaternionFunctionOfTime<deriv_order>&>(
+      *(cache_f_of_t_map.at(quatfot_name)));
+
+  CHECK(cache_pp == expected_pp);
+  CHECK(cache_quatfot == expected_quatfot);
+
+  // Update functions of time in global cache with new expiration time
+  const double newer_expiration_time = new_expiration_time + 1.0;
+  for (auto& name : {pp_name, quatfot_name}) {
+    Parallel::mutate<domain::Tags::FunctionsOfTime,
+                     control_system::Mutations::
+                         reset_function_of_time_expiration_time<deriv_order>>(
+        cache, name, newer_expiration_time);
+  }
+
+  // Update expected function of time
+  expected_pp.reset_expiration_time(newer_expiration_time);
+  expected_quatfot.reset_expiration_time(newer_expiration_time);
+
+  const auto& cache_pp2 = dynamic_cast<
+      const domain::FunctionsOfTime::PiecewisePolynomial<deriv_order>&>(
+      *(cache_f_of_t_map.at(pp_name)));
+  const auto& cache_quatfot2 = dynamic_cast<
+      const domain::FunctionsOfTime::QuaternionFunctionOfTime<deriv_order>&>(
+      *(cache_f_of_t_map.at(quatfot_name)));
+
+  // Check that the FunctionsOfTime and expiration times are what we expected
+  CHECK(cache_pp2 == expected_pp);
+  CHECK(cache_quatfot2 == expected_quatfot);
+  CHECK(cache_pp2.time_bounds()[1] == newer_expiration_time);
```

Comment:
Just verifying: If this fails the check two lines above should also have failed, correct?  (I am not asking you to remove this check.)

Reply:
Yes I believe so. PiecewisePolynomial checks that the expiration times are equal and `time_bounds()[1]` is the stored expiration time.